src/ripple/protocol/impl/RippleAddress.cpp

```diff
@@ -602,22 +587,10 @@ bool RippleAddress::accountPrivateSign (uint256 const& uHash, Blob& vucSig) cons
 Blob RippleAddress::accountPrivateEncrypt (
     RippleAddress const& naPublicTo, Blob const& vucPlainText) const
 {
-    openssl::ec_key secretKey = ECDSAPrivateKey (getAccountPrivate());
-    openssl::ec_key publicKey = ECDSAPublicKey (naPublicTo.getAccountPublic());
-    
-    Blob vucCipherText;
+    uint256 secretKey = getAccountPrivate();
+    Blob    publicKey = naPublicTo.getAccountPublic();
```

Comment:
Does it makes sense to have a "secret key type" and a "public key type" instead of uint256 and Blob? (I'm sure this code change would ripple through the codebase, so I'm not seriously proposing you change).